src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTests.cs

```diff
@@ -3175,5 +3175,202 @@ public class LambdaGenerator(Action<IncrementalGeneratorInitializationContext> a
         {
             public void Initialize(IncrementalGeneratorInitializationContext context) => action(context);
         }
+
+        [Fact]
+        public async Task IncrementalCompilation_NothingRuns_When_AdditionalFiles_HaveSameContent()
+        {
+            // Arrange
+            using var eventListener = new RazorEventListener();
+            var project = CreateTestProject(new()
+            {
+                ["Pages/Index.razor"] = "<h1>Hello world</h1>",
+                ["Pages/Counter.razor"] = "<h1>Counter</h1>",
+            });
+            var compilation = await project.GetCompilationAsync();
+            var (driver, additionalTexts) = await GetDriverWithAdditionalTextAsync(project);
+
+            var result = RunGenerator(compilation!, ref driver)
+                            .VerifyPageOutput(
+@"#pragma checksum ""Pages/Index.razor"" ""{ff1816ec-aa5e-4d10-87f7-6f4963833460}"" ""6b5db227a6aa2228c777b0771108b184b1fc5df3""
```

Comment:
Could we save this page output which is the same across many tests in some common variable so we don't have to update so many tests when we change codegen?

Alternatively, I think we might not need to verify page output in these new tests at all since their purpose is not to verify the codegen but rather the incrementality of the source generator.

Reply:
I just removed the strings, as we aren't actually testing the codegen here. Agree we should consider moving these tests over to external baselines though.